src/compiler/checker.ts

```diff
@@ -46808,6 +46812,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         if (checkExternalImportOrExportDeclaration(node)) {
             const importClause = node.importClause;
+            const moduleExisted = resolveExternalModuleName(node, node.moduleSpecifier);

```

Comment:
After the fix, some errors went away. Upon further inspection, I found that this code path was not resolving module names for specifier-less imports like:
```ts
import 'stuff'
```

So to bring back the errors that were already there I fixed this. One thing to notice about this is that those errors were reported **inconsistently**. They were reported for incremental/composite builds as that led to resolving everything to build the program's state and its `referenceMap`. They were not reported for normal builds.

However, with many baseline changes here we can see that this could be considered a breaking change. I think it would be a good "breaking change" here. The fact that the errors were reported inconsistently was weird and I doubt it was desired.

To quote @weswigham from https://github.com/microsoft/TypeScript/pull/28558 :
> Although I really do think it's strange we don't error on these kinds of imports.

Reply:
Yee, I have some Babel background and those are specifiers there: https://babeljs.io/docs/babel-types#importdeclaration ;p

> Not erroring on side-effect-only imports that don't resolve is intentional and issues are indeed reported about it from time to time. Explanation of the behavior is here: https://github.com/microsoft/TypeScript/issues/50394#issuecomment-1381014487

Thanks for the pointers. I think that, at the very least, we should issue errors on `import './foo.ts'` with `allowImportingTsExtensions: false`. That is what originally prompted me to introduce this change - such an error was removed from the composite test suite.

In addition to that, it feels like `import 'foo'` is also questionable... as the user, I would really like to get an error on this if I forgot to install `foo`. In some sense, I feel like this could only allow such side-effects imports for unknown extensions.

Also, this current behavior feels weird:
```ts
import 'foo' // no error
import {} from 'foo' // error
```

Reply:
Huh, Babel is being weird here because in ECMAScript the thing called "specifier" is very specifically (no pun intended) the module name:

Reply:
I don't know what the initial source of this terminology is, but all ASTs seem to use `source` for the thing being imported from and `specifiers` for the thing(s) being imported (e.g. check `import foo from 'bar'` on https://astexplorer.net/)

Reply:
Not responding these types of imports is a designed behavior, fwiw. I don't like it and want to change it too, but we'd break everyone importing css files and so on. So I don't think this is in scope for a bugfix PR

Reply:
I reverted those changes from here